### PR TITLE
Fetch

### PR DESCRIPTION
**Pull request by PetalCat**
## Description

~~Pretend to~~Actually fetch the number from Appwrite.

_Closes: #9_

## Type of Change

- 🧹 Code refactor

### CODE REVIEW

**Review by lishaduck — DISMISSED** [review]

**Comment by lishaduck** [comment]
https://github.com/PSDTools/app/blob/72a7b209a768d1e5fdcb91359aba815fc4245662/lib/app/view/app.dart#L7
@PetalCat, can you remove this line?

**Review by lishaduck — CHANGES_REQUESTED** [review]
https://github.com/PSDTools/app/blob/72a7b209a768d1e5fdcb91359aba815fc4245662/lib/app/view/app.dart#L7
@PetalCat, can you remove this line?

File: lib/features/pirate_coins/domain/coins_domain.dart [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
+import "package:riverpod_annotation/riverpod_annotation.dart";
+
+import "../data/coins_data.dart";
+
 part "coins_domain.freezed.dart";
+part "coins_domain.g.dart";
 
+/// A number.
 @freezed
 abstract class NumberDomain with _$NumberDomain {
```
Comment (lishaduck) [inline lib/features/pirate_coins/domain/coins_domain.dart:11]: Is there a better name for this? I'm thinking `Coins`. It needs to keep being usable with more data for it to hold added.

Comment (PetalCat) [reply lib/features/pirate_coins/domain/coins_domain.dart:11]: Yeah... I agree... What should I rename where

Comment (lishaduck) [reply lib/features/pirate_coins/domain/coins_domain.dart:11]: F2 it.

Comment (lishaduck) [reply lib/features/pirate_coins/domain/coins_domain.dart]: Better name: `CoinsModel`.

Comment (lishaduck) [reply lib/features/pirate_coins/domain/coins_domain.dart]: Also *note to self* I need to make `API` an `@freezed` class.

File: lib/features/pirate_coins/domain/coins_domain.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+/// Get coins data from data layer.
+@riverpod
+class CoinsDomain extends _$CoinsDomain {
+  @override
+  NumberDomain build() {
+    final number = ref.watch(coinsDataProvider);
+    return NumberDomain(number: number);
+  }
```
Comment (lishaduck) [inline lib/features/pirate_coins/domain/coins_domain.dart:27]:
```suggestion
class CoinsModel with _$CoinsModel {
  /// Create a new, immutable instance of [CoinsModel ].
  const factory CoinsModel ({
    /// The number.
    required AsyncValue<int> coins,
  }) = _CoinsModel;
}

/// Get coins data from data layer.
@riverpod
class Coins extends _$Coins {
  @override
  CoinsModel build() {
    final number = ref.watch(coinsDataProvider);
    return CoinsModel(coins: coins);
  }
```

File: lib/features/utils/presentation/device_info_dialog.dart [hunk 3]
```diff
@@ -19,7 +19,7 @@ class DeviceInfoDialog extends ConsumerWidget {
 
     return AlertDialog(
       title: Container(
-        padding: const EdgeInsets.all(15),
+        padding: const EdgeInsets.all(150),
```
Comment (lishaduck) [inline lib/features/utils/presentation/device_info_dialog.dart:22]: Do we know that this works?

File: lib/utils/data/api.dart [hunk 4]
```diff
@@ -14,17 +14,35 @@ class Api implements ApiRepository {
   Api({
```
Comment (lishaduck) [inline lib/utils/data/api.dart:14]: This class would be better as an `@freezed` class. Or something.

File: lib/utils/data/secrets.dart [hunk 5]
```diff
@@ -18,3 +18,15 @@ String projectId(ProjectIdRef _) => Dartdefine.projectId;
...
+@riverpod
+String databaseId(DatabaseIdRef _) => Dartdefine.databaseId;
+
+/// Get the Appwrite collection ID.
+@riverpod
+String collectionId(CollectionIdRef _) => Dartdefine.collectionId;
+
+/// Get the Appwrite document ID.
+@riverpod
+String documentId(DocumentIdRef _) => Dartdefine.documentId;
```
Comment (lishaduck) [inline lib/utils/data/secrets.dart:32]: Still think there's got to be a better way to do this. Maybe an enum w/ a case statement?

**Review by lishaduck — CHANGES_REQUESTED** [review]

File: lib/features/utils/presentation/device_info_dialog.dart (hunk 3 above)
Comment (lishaduck) [reply lib/features/utils/presentation/device_info_dialog.dart:22]: It doesn't.

File: lib/features/pirate_coins/domain/coins_domain.dart (hunk 2 above)
Comment (lishaduck) [reply lib/features/pirate_coins/domain/coins_domain.dart]: 0c67e624376f65e40d6ccd697208de4ea4079bb6 fixed this.

File: lib/utils/data/secrets.dart (hunk 5 above)
Comment (lishaduck) [reply lib/utils/data/secrets.dart:32]: See #112.

**Review by lishaduck — APPROVED** [review]

**Pull request merged**